lib/DxilPIXPasses/DxilPixEmitResources.cpp

```diff
@@ -0,0 +1,43 @@
+///////////////////////////////////////////////////////////////////////////////
+//                                                                           //
+// DxilPixEmitResources.cpp                                                  //
+// Copyright (C) Microsoft Corporation. All rights reserved.                 //
+// This file is distributed under the University of Illinois Open Source     //
+// License. See LICENSE.TXT for details.                                     //
+//                                                                           //
+// Emit Dxil resource meta data                                              //
+//                                                                           //
+///////////////////////////////////////////////////////////////////////////////
+
+#include "dxc/DXIL/DxilFunctionProps.h"
+#include "dxc/DXIL/DxilModule.h"
+#include "dxc/DXIL/DxilOperations.h"
+#include "dxc/DXIL/DxilUtil.h"
+#include "dxc/DxilPIXPasses/DxilPIXPasses.h"
+#include "dxc/DxilPIXPasses/DxilPIXVirtualRegisters.h"
+#include "dxc/HLSL/DxilGenerationPass.h"
+
+#include "llvm/IR/Module.h"
+
+using namespace llvm;
+using namespace hlsl;
+
+class EmitDxilResourceToMeta : public ModulePass {
+public:
+  static char ID;
+  explicit EmitDxilResourceToMeta() : ModulePass(ID) {}
+  bool runOnModule(Module &M) override {
+    DxilModule &DM = M.GetOrCreateDxilModule();
+    DM.ReEmitDxilResources();
```

Comment:
I find the name of this is misleading.  Originally, when named, it was for the purpose of re-emitting resource metadata, but now it re-emits *all* metadata.  The same issue extends to this pass name and the descriptions of the pass.  I think it all should be changed to indicate that all metadata is being re-emitted.  The change could be made first in just the names and descriptions around this pass, with a follow-up to change the name of the method in DxilModule, since that will have broader impact.  Or it all could be done at the same time, if it's not too much of a pain.

Reply:
Changed the pass name. Will change the fn in later PR as suggested.